sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/services/NewOpenAiService.java

```diff
@@ -0,0 +1,178 @@
+import java.util.Map;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+/** Service class for OpenAI service */
+@Service
+@Slf4j
+public class NewOpenAiService {
+
+  /**
+   * Chat request to OpenAI
+   *
+   * @param prompt The prompt to send to the assistant
+   * @return the assistant message response
+   */
+  @Nonnull
+  public OpenAiChatCompletionResponse chatCompletion(@Nonnull final String prompt) {
+    return OpenAiClient.forModel(GPT_35_TURBO)
+        .chatCompletion(new OpenAiChatCompletionRequest(prompt));
+  }
+
+  /**
+   * Asynchronous stream of an OpenAI chat request
+   *
+   * @return the emitter that streams the assistant message response
+   */
+  @Nonnull
+  public Stream<OpenAiChatCompletionDelta> streamChatCompletionDeltas(
+      @Nonnull final String message) {
+    final var request = new OpenAiChatCompletionRequest(OpenAiMessage.user(message));
+
+    return OpenAiClient.forModel(GPT_35_TURBO).streamChatCompletionDeltas(request);
+  }
+
+  /**
+   * Asynchronous stream of an OpenAI chat request
+   *
+   * @return the emitter that streams the assistant message response
+   */
+  @Nonnull
+  public Stream<String> streamChatCompletion(@Nonnull final String message) {
+    return OpenAiClient.forModel(GPT_35_TURBO)
+        .withSystemPrompt("Be a good, honest AI and answer the following question:")
+        .streamChatCompletion(message);
+  }
+
+  /**
+   * Chat request to OpenAI with an image
+   *
+   * @param linkToImage The link to the image
+   * @return the assistant message response
+   */
+  @Nonnull
+  public CreateChatCompletionResponse chatCompletionImage(@Nonnull final String linkToImage) {
+    final var partText =
+        new ChatCompletionRequestMessageContentPartText()
+            .type(TEXT)
+            .text("Describe the following image.");
+    final var partImageUrl =
+        new ChatCompletionRequestMessageContentPartImageImageUrl()
+            .url(URI.create(linkToImage))
+            .detail(HIGH);
+    final var partImage =
+        new ChatCompletionRequestMessageContentPartImage().type(IMAGE_URL).imageUrl(partImageUrl);
+    final var userMessage =
+        new ChatCompletionRequestUserMessage()
+            .role(USER)
+            .content(ChatCompletionRequestUserMessageContent.create(List.of(partText, partImage)));
+    final var request =
+        new CreateChatCompletionRequest()
+            .addMessagesItem(userMessage)
+            .functions(null)
+            .tools(null)
+            .parallelToolCalls(null);
+
+    return OpenAiClient.forModel(GPT_4O).chatCompletion(request);
+  }
+
+  /**
+   * Chat request to OpenAI with a tool.
+   *
+   * @param description of the function to be sent to the assistant
+   * @return the assistant message response
+   */
+  @Nonnull
+  public OpenAiChatCompletionResponse chatCompletionTools(@Nonnull final String description) {
```

Comment:
Please remove the parameter, the description of the Fibonacci tool is fixed anyway

Reply:
I agree.. the parameter by itself doesn't make too much sense.

We are looking at a "service" class here. That's a middle-layer between webapp backend and external services. We need to think about what could be a "realistic" API for that middle layer. Having no parameter would be counter-intuitive to me, for such a middle-layer.

Maybe we can change the service method here to consider the `int months` as parameter:
```diff
- How many rabbits will there be after 12 months")
+ How many rabbits will there be after %s months?".formatted(months))
```